misk-redis/src/main/kotlin/misk/redis/Redis.kt

```diff
@@ -721,3 +753,144 @@ inline fun checkHrandFieldCount(count: Long) {
     "You must request at least 1 field."
   }
 }
+
+/**
+ * Like [Redis], but returns [Supplier]s to defer value retrieval.
+ * Does not support transactions or pubsub.
+ */
+interface DeferredRedis {
+  fun del(key: String): Supplier<Boolean>
```

Comment:
I originally thought of making these suspend funs that return Deferred, but I think Suppliers are better for misk's threading model?

Would appreciate other's thoughts here.